### DISCUSSION TITLE
TaxonomyField

### DISCUSSION THREAD

**Discussion by asemandeveloper** [OrchardCMS/OrchardCore#12700] [General]
How can I create a taxonomy field by coding in DataMigration?

**Comment by hishamco** [comment]
You can add it like other fields

```csharp
part.Alter<TaxonomyField>("Tags", tf =>
{

});
```

Or we could use it as the following:

```csharp
_contentDefinitionManager.AlterPartDefinition("Article", part => part
    .WithField("Tags", field => field.OfType("TaxonomyField")
);
```

**Comment by asemandeveloper** [comment]
Hi hishamco
Thank you for your guidance
My code is similar to below.

    public int Create()
    {
        _contentDefinitionManager.AlterTypeDefinition("Assistant", type => type
            .WithPart("TitlePart")
            .WithPart("Autoroute")
        );
        _contentDefinitionManager.AlterPartDefinition("WorkshopPart", part => part
            .Attachable()
            .WithField("Assistants", field => field
                .OfType(nameof(TaxonomyField))
                .WithSettings(new ContentPartFieldSettings()
                {
                    DisplayName = "Assistants",
                    Editor = "Assistants",
                    DisplayMode = "Assistants"
                })
            )
        );
        return 1;
    }

    public async Task<int> UpdateFrom1Async()
    {
        await CreateAssistantContentItemAsync();
        return 2;
    }

    private async Task CreateAssistantContentItemAsync()
    {
        var taxonomy = await _contentManager.NewAsync("Taxonomy");
        taxonomy.DisplayText = "Assistants";
        var taxonomyPart = taxonomy.As<TaxonomyPart>();
        taxonomyPart.TermContentType = "Assistant";
        taxonomy = taxonomy.Apply(taxonomyPart);
        await _contentManager.CreateAsync(taxonomy, VersionOptions.Published);
    }
}


**I receive the following message in the view**

![image](https://user-images.githubusercontent.com/13555275/197587940-430abaf5-e81b-470b-a687-921085985cf2.png)

**Reply by hishamco** [reply]
Disable then enable your module to let the data migration execute

**Reply by asemandeveloper** [reply]
Thank you for your reply
I did this, but I get the same message!
